wrapper/src/main/java/software/amazon/jdbc/plugin/IamAuthConnectionPlugin.java

```diff
@@ -154,7 +155,47 @@ private Connection connectInternal(String driverProtocol, HostSpec hostSpec, Pro
           cacheKey,
           new TokenInfo(token, Instant.now().plus(tokenExpirationSec, ChronoUnit.SECONDS)));
     }
-    return connectFunc.call();
+
+    try {
+      return connectFunc.call();
+    } catch (final SQLException exception) {
+
+      LOGGER.finest(
+          () -> Messages.get(
+              "IamAuthConnectionPlugin.connectException",
+              new Object[] {exception}));
+
+      if (!this.pluginService.isLoginException(exception) || !isCachedToken) {
+        throw exception;
```

Comment:
Agreed with Aaron, can we log the token and `isCachedToken` if it is a login exception. Would be beneficial for debugging purposes